proxy-server/go/proxy-main.go

```diff
@@ -0,0 +1,306 @@
+package main
+
+import (
+	"crypto/rand"
+	"fmt"
+	"github.com/dgrijalva/jwt-go"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+)
+
+//SECURITY NOTICE: Some older versions of Go have a security issue in the cryotp/elliptic.
+//Recommendation is to upgrade to at least 1.8.3.
+
+var defaultURL = "http://localhost"           // resource default url. port 80 of the localhost
+var fallbackURL = "http://localhost/fallback" // not authorized fallback
+var jwtKey []byte
+var sharedKey []byte
+
+var minutesExpire = 60 //  1 hour to expiration
+var debug = true
+var defaultPort = "8080"
+var mutex sync.Mutex
+var tokenMap = make(map[string]string)
+
+const keyLength = 512 / 8
+
+const (
+	errGeneric          = iota
+	errNoCookie         = iota
+	errBadSignature     = iota
+	errExpired          = iota
+	errTokenInvalid     = iota
+	errSignatureInvalid = iota
```

Comment:
you can omit the `iota` after all but the first of these. https://github.com/golang/go/wiki/Iota